src/main/java/service/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,27 @@
+package service;
+
+import dao.Storage;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private static final String FRUIT_QUANTITY_LABEL = "Fruit, Quantity";
+    private static final String SEPARATOR = ", ";
+    private final Storage storage;
+
+    public ReportGeneratorImpl(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public String generateReport(Map<String, Integer> fruitQuantities) {
+        Map<String, Integer> fruitQuantitiesFromStorage = storage.getAllFruitsWithQuantity();
+
+        StringBuilder report = new StringBuilder();
+        report.append(FRUIT_QUANTITY_LABEL).append(System.lineSeparator());
```

Comment:
```suggestion
        StringBuilder report = new StringBuilder()
                    .append(FRUIT_QUANTITY_LABEL).append(System.lineSeparator());
```
Not a mistake but FYI